### DISCUSSION TITLE
2fa, totp

### DISCUSSION THREAD

**Discussion by tabatip** [kimai/kimai#3804] [On-Premise: Installation & Updates]
Dear all,

I'm currently investigate if kimai2 is the right solution for us in company. It look pretty cool, most of my requirements are fine solved. We are currently migrating our lot of distributed cloud services to some on-prem or at least private server environments.

The documentation is mostly clear and easy. Thank you, Congrats for effort. What I missing: TOTP how-to. I see - its supported, but cannot find how. Is there anywhere a description about? 

Thanks
tabatip

**Comment by tabatip** [comment]
Dear all,

Sorry for noise. Self answering: Switch the language to english or german - and "magially" appers in menu. ( in user settings) -info: I use a non-very-fequently usen language as base :) 

Thanks
tabatip

**Reply by kevinpapst** [reply]
Which language do you use? Translations should always fallback to english if they are missing.

**Comment by mase1981** [comment]
@tabatip can you share where you found this? i just installed kimai and i am set to english , but still cannot find where to enable 2FA in any setting. Can you please share where you found the feature to enable it? TIA.

**Reply by kevinpapst** [reply]
2FA is only available in the 2.0 branch, which is not yet officially released.
But you can install it or upgrade to it by using 2.0.0 as tag.

**Reply by mase1981** [reply]
Got it, yeah i am on 1.30.5 stable (Ayumi), so that make sense. thank you for clarifying it ! is there an ETA for the 2.0 official release? i can upgrade but prefer to have teh stable release as i intend to use kimai officially.

**Reply by kevinpapst** [reply]
The tag is there. The more people test, the more stable 2.0 will become.
The more people think staying on 1.30 is wiser, the longer it takes 🤷

**Reply by mase1981** [reply]
Whelp @kevinpapst you make perfect sense 💯  :) Sorry about that ! :) Just to confirm, when you say "tag" you mean pull the specific tag image from docker? https://hub.docker.com/r/kimai/kimai2/tags , if so, which tag in particular will be for this 2.0 version ? Thank you and i truly appreciate it.

**Reply by tabatip** [reply]
Hi, I'm using Hungarian language.

Now I had reinstalled again on clean environment, without any prev database,etc. - to test again. Result: I cannot predict, when will show up the totp menu on first login - in HU language. Sometimes is it there, somtimes comes up when If I switch to any other menu and back to user edit.  Important - if the menu was there every time there :) so this is a "startup" or "user creation" issue.  

 The English language does not have this small issue. If I will have a little bit more time I will look into logs, but so small is the point, that wasn't yet worth to spend resource for ( My first assumption something with charset, which is special in lang_hu )

**Reply by kevinpapst** [reply]
Not able to reproduce it, it just appeared directly with `hu_HU` as locale (and also with `hu` only):
<img width="610" alt="Bildschirm­foto 2023-02-16 um 14 00 19" src="https://user-images.githubusercontent.com/533162/219371661-c40891b8-dd21-4ed1-8dcb-b0b547e8c33d.png">

It is not translated, but Kimai uses english as fallback in that case.

You could help to translate Kimai and this key in particular: 
https://hosted.weblate.org/translate/kimai/messages/hu/?checksum=97f177571b82fad9

Please note, that 2FA is only available if the suer is registered via database. If login is configured through LDAP or SAML, 2FA is not available to that user account!
